services/grpc/grpc.js

```diff
@@ -248,6 +248,42 @@ class ZapGrpc extends EventEmitter {
 
     return { host, cert, macaroon, waitForMacaroon, waitForCert, useMacaroon, protoDir }
   }
+
+  /**
+   * async subscribeChannelGraph - Set up subscription to the channel graph stream.
+   *
+   * There is no guarentee that it is ready yet as it can take time for lnd to start it once chain sync has finished
+   * so set up a schedular to keep retrying until it works.
+   */
+  subscribeChannelGraph() {
+    const backoff = 2
+    const maxTimeout = 1000 * 60
+    let timeout = 250
+
+    const initSubscription = async () => {
+      if (this.grpc.state !== 'active') {
+        return
+      }
+
+      this.subscribe('channelgraph')
+
+      // If the channel graph subscription fails to start, try again in a bit.
+      if (this.activeSubscriptions.channelgraph) {
+        this.activeSubscriptions.channelgraph.once('error', async error => {
+          if (error.message === 'router not started') {
+            grpcLog.warn('Unable to subscribe to channelgraph. Will try again in %sms', timeout)
+            await delay(timeout)
+            initSubscription()
```

Comment:
Is it possible to use `scheduler` instead of managing backoff manually?

Reply:
I originally tried to do it that way, but the issue is that this isn't quite a straightforward as retrying something on a backoff schedule as here there is an actually trigger other than the time based trigger - we only want to schedule a retry if we were unsuccessful in our previous subscribe attempt. ie, we are using the error event to trigger scheduling of future retry.

Reply:
Makes sense. What do you think if we extend scheduler and add the ability to pause task execution? You will pause on success and resume on error. This way the backoff schedule will be maintained. The initial version of scheduler was retryOnError util, which has transformed into scheduler. This pattern will probably prove useful in future cases too.

Reply:
I'm not sure that would work either, since there is no notable success case here - success is just the absence of an error event.